includes/classes/MeasurementUnits.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Zencart;
+
+use TypeError;
+
+class MeasurementUnits
+{
+    /**
+     * Convert weight from one unit to another
+     */
+    public static function convertWeight(float|int $incoming, string $from_unit, string $to_unit, ?int $precision): float|int
+    {
+        if (!in_array($from_unit, ['lbs', 'kgs', 'oz', 'g'])) {
+            throw new TypeError('Invalid weight unit for $from_unit: ' . $from_unit);
+        }
+        if (!in_array($to_unit, ['lbs', 'kgs', 'oz', 'g'])) {
+            throw new TypeError('Invalid weight unit for $to_unit: ' . $to_unit);
+        }
+
+        $ratio = [];
+        $ratio['lbs']['oz'] = 16;
+        $ratio['lbs']['kgs'] = 0.45359237;
+        $ratio['lbs']['g'] = 453.59237;
+        $ratio['oz']['lbs'] = 0.0625;
+        $ratio['oz']['kgs'] = 0.02834952; //0.0625 * 0.45359237;
+        $ratio['oz']['g'] = 28.34952; //0.0625 * 0.45359237 * 1000;
+        $ratio['kgs']['oz'] = 35.2739619; //2.20462262 * 16;
+        $ratio['kgs']['lbs'] = 2.20462262;
+        $ratio['kgs']['g'] = 1000;
+        $ratio['g']['oz'] = 0.0352739619; //0.001 * 2.20462262 * 16;
+        $ratio['g']['kgs'] = 0.001;
+        $ratio['g']['lbs'] = .00220462262;
+
+        if ($from_unit !== $to_unit) {
+            $result = $incoming * $ratio[$from_unit][$to_unit];
+        } else {
+            // if converting "to" same as "from", then use original value and carry on to rounding
+            $result = $incoming;
+        }
+
+        if (($precision === null) && in_array($to_unit, ['lbs', 'kgs'])) {
```

Comment:
Just curious, why limit this precision override to lbs and kgs?

Reply:
The feeling was that grams/ounces might need extra precision, but converting to lbs/kgs likely doesn't need 0.0000001 accuracy because they're already such larger units. 
Also, it was only with lbs/kgs that this same thing was done in another mod I'd looked at for additional input.